homeassistant/components/honeywell/climate.py

```diff
@@ -97,6 +98,37 @@ async def async_setup_entry(
             for device in data.devices.values()
         ]
     )
+    remove_stale_devices(hass, entry, data.devices)
+
+
+def remove_stale_devices(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    devices: dict[str, SomeComfortDevice],
+) -> None:
+    """Remove stale devices from device registry."""
+    device_registry = dr.async_get(hass)
+    device_entries = dr.async_entries_for_config_entry(
+        device_registry, config_entry.entry_id
+    )
+    all_device_ids: list = []
+    for device in devices.values():
+        all_device_ids.append(device.deviceid)
+
+    for device_entry in device_entries:
+        device_id: str | None = None
+
+        for identifier in device_entry.identifiers:
+            device_id = identifier[1]
```

Comment:
I think we should make sure that the first item in the identifier matches the honeywell domain. I device can generally be connected to more than a single integration.

Reply:
Here's an example:
https://github.com/home-assistant/core/blob/684b2d45370813834c5dff613740d3a0a124f3ce/homeassistant/components/tradfri/__init__.py#L154-L193

Reply:
I was thinking only this domain would be attached - addressed in separate PR